### ISSUE TITLE
Downloading a video with comments in their own file and not in info.json

### ISSUE THREAD

**Issue by Aces-and-Jacks** [yt-dlp/yt-dlp#11240]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

I was wondering if comments can be added into their own file, and if so can someone help me with it? 

I know `--write-comments` would download them into `video.info.json` however I want to have them write to a new `comments.json` file. I tried following along with #2372 and #455 but it got too compicated for me to understand. I tried pasting from [this](https://github.com/yt-dlp/yt-dlp/issues/2372#issuecomment-2102765808) `--write-comments --print-to-file "after_filter:%(comments)j" "comments.json" --parse-meta "video::( P<comments>)"` but it didn't create anything resembilng a `comments.json` file. I figure blindly copy and pasting a command probably wouldn't help but I couldn't understand it. 

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by Aces-and-Jacks** [comment]
So basically what would I need to put in my config to make this work?

**Comment by bashonly** [comment]
> `--write-comments --print-to-file "after_filter:%(comments)j" "comments.json" --parse-meta "video::( P<comments>)"` but it didn't create anything resembilng a `comments.json` file

You're missing a `?` before the `P` in `(?P<comments>)`, but otherwise this should work. If not, please give details, share a verbose log, share the JSON file

**Comment by Aces-and-Jacks** [comment]
share `video.info.json` file? Also I don't know what the verbose log would be. Nor what details you may need. 

I'm on macOS, last I checked I'm on latest yt-dlp. and I fixed the command with `--write-comments --print-to-file "after_filter:%(comments)j" "comments.json" --parse-meta "video::(?P<comments>)"`

**Comment by Aces-and-Jacks** [comment]
This is what terminal showed https://mclo.gs/N6tre77

**Comment by bashonly** [comment]
What's wrong with the `comments.json` file?

**Comment by Aces-and-Jacks** [comment]
It didn't create it

**Comment by bashonly** [comment]
```
[info] Writing '%(comments)j' to: comments.json
```

**Comment by Aces-and-Jacks** [comment]
yeah, I saw that. Which is why I have no idea why it didn't create it

**Comment by bashonly** [comment]
it created it in your terminal's current working directory

**Comment by Aces-and-Jacks** [comment]
huh...you know that's one thing I didn't try. searching for the file with a file browser. It creates it in my home directory but how would I make it go into a folder I want?

**Comment by Aces-and-Jacks** [comment]
this is the location `'/Users/$USER/Library/Application Support/Stacher/Downloads/%(uploader)s-[%(uploader_id)s]/%(uploader)s-%(upload_date>%Y-%m-%d)s_%(title)s-[%(id)s]/%(uploader)s-%(upload_date)s_%(title)s-[%(id)s].%(ext)s' `

**Comment by bashonly** [comment]
instead of `--print-to-file "after_filter:%(comments)j" "comments.json"` use `--print-to-file "after_filter:%(comments)j" "/Users/username/Library/Application Support/Stacher/Downloads/comments.json"` or whatever absolute path you want

**Comment by Aces-and-Jacks** [comment]
Thank you so much for your help I appreciate it a lot, and that worked perfectly. And sorry for the late reply, I was tweaking and testing a bit. 

I did 
`--write-comments --print-to-file "after_filter:%(comments)j" "/Users/$USER/Library/Application Support/Stacher/Downloads/%(uploader)s-[%(uploader_id)s]/%(uploader)s-%(upload_date>%Y-%m-%d)s_%(title)s-[%(id)s]/%(uploader)s-%(upload_date)s_%(title)s-[%(id)s].comments.json" --parse-meta "video::(?P<comments>)"`

**Comment by Aces-and-Jacks** [comment]
Hopefully this'd help anyone else who wants to do this

**Comment by Aces-and-Jacks** [comment]
And do you know if that'd remove comments from the `video.info.json` file?

**Comment by bashonly** [comment]
> And do you know if that'd remove comments from the `video.info.json` file?

That's what the `--parse-meta "video::(?P<comments>)"` part does

**Comment by Aces-and-Jacks** [comment]
Ah, ok. Well thank you again for your time

**Comment by barkoder** [comment]
How do you limit the number of comment pages that yt-dlp crawls? Say I only want 5 pages.

What should I add to `--write-comments --print-to-file "after_filter:%(comments)j" "%(uploader)s-%(upload_date)s-%(title)s-%(id)s.comments.json" --parse-meta "video::(?P<comments>)"`

to make that happen?

**Comment by bashonly** [comment]
@barkoder use the `max_comments` youtube extractor-arg with the `--extractor-args` option

https://github.com/yt-dlp/yt-dlp?tab=readme-ov-file#youtube

> * `max_comments`: Limit the amount of comments to gather. Comma-separated list of integers representing `max-comments,max-parents,max-replies,max-replies-per-thread`. Default is `all,all,all,all`
    * E.g. `all,all,1000,10` will get a maximum of 1000 replies total, with up to 10 replies per thread. `1000,all,100` will get a maximum of 1000 comments, with a maximum of 100 replies total

**Comment by Xylemon** [comment]
Apologies for bumping an old closed issue, but I don't know if this is worth opening a whole new issue:

A problem I have been having with external comment downloading is that an existing comments file will be redownloaded. While I can bash script my way into checking for an individual video's existing comment file exists, I have no idea how to check for a whole channel (which is something I require for keeping certain channels up to date in automated fashion).  Plus it would be better if it was just auto detected by YT-DLP itself and did the "right thing". Here is my command which has served me well:

`--no-download-archive --no-overwrite --skip-download --get-comments --print-to-file "\%\(comments\)j" "\%\(id\)s.comments.json"`

Any ideas on what I need to add to make sure YT-DLP will automatically detect an existing `idhere.comments.json` file and not overwrite it? It seems --no-overwrite does not account for this behavior (unfortunately).

**Comment by barkoder** [comment]
@Xylemon Add `%(epoch>%Y%m%d%H%M%S)s` to your output comments.json wherever you prefer.

This will output the filename with the current date and time in the format - YYYYMMDDHHMMSS - 20251009210940

Read https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes

You could also add microseconds `%f` to your filename if you want. ;)

Cheers.

**Comment by Xylemon** [comment]
> [@Xylemon](https://github.com/Xylemon) Add `%(epoch>%Y%m%d%H%M%S)s` to your output comments.json wherever you prefer.
> 
> This will output the filename with the current date and time in the format - YYYYMMDDHHMMSS - 20251009210940
> 
> Read https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes
> 
> You could also add microseconds `%f` to your filename if you want. ;)
> 
> Cheers.

I appreciate the reply, but this doesn't solve my problem. I am trying to make it skip comment downloading altogether when a comment file already exists for the detected video.  For context, I automate channel crawling (and playlists) with yt-dlp, so it downloading a newer copy of a comment file not only adds to the time processes and space on my HDD, but will also add to the amount of viewable pages for that hour/day on the account.

I have tried to make it part of the normal download routine, but it seems yt-dlp will always inject comments into `info.json` with `--get-comments` unless you forcibly run it a second time with the command I posted above. Since it has to be ran a second time, `--download-archive` kicks in and will skip comments.

This would all be so much easier if yt-dlp did not force comments into `info.json` or at least made external comment files a feature. Some videos have 1,000,000+ comments leading to a 150+ MB `info.json` which is harder to query, and it's also easier to see if you have comments for said video (instead of some jq shenanigans that now has to query _every single_ json). Comments have also had formatting issues in the past, this would lead to corrupted `info.json` files, again, undesirable behavior. At least when they are in a separate file, you still get the benefit of that metadata being preserved yet standalone.

If there is some way I can detect an already existing comments file within the current version, I would love to know. Otherwise I will have to do some very complex `diff` trick with the `--download-archive` file, which is not desirable.
